lms/models/region.py

```diff
@@ -18,25 +18,21 @@ class Region:
 class Regions:
     """A collection of all the regions as an enum like object."""
 
-    # As region objects are frozen, these can be directly compared like enums
-    US = Region(code="us", name="Worldwide (U.S.)", authority="lms.hypothes.is")
-    CA = Region(code="ca", name="Canada", authority="lms.ca.hypothes.is")
-    ALL = US, CA
-
-    _AUTHORITY_MAP = {region.authority: region for region in ALL}
-    _CODE_MAP = {region.code: region for region in ALL}
-
     @classmethod
-    def from_code(cls, code: str) -> Region:
+    def from_code(cls, authority: str, code: str) -> Region:
         """
         Get a region object based on its code.
 
         :raises ValueError: If no valid region can be found.
         """
+        names = {"us": "Worldwide (U.S.)", "ca": "Canada"}
+
         try:
-            return cls._CODE_MAP[code]
+            name = names[code]
         except KeyError as err:
-            raise ValueError(f"Cannot find a region for the code '{code}'") from err
+            raise ValueError(f"Cannot find a name for the code '{code}'") from err
+
+        return Region(code=code, name=name, authority=authority)
```

Comment:
`from_code()` now constructs a `Region` from the given argument, rather than returning one of the two pre-constructed regions.